### PR TITLE
You'll see how cool this is

### PR DESCRIPTION
**Pull request by Tabardelli**

### CODE REVIEW

**Comment by Raziel-777** [comment]
Warning you push useless files (.idea)

File: java/src/production/ConvertCsvInList.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        // open file input stream
+        int manualId = 1;
+        String line;
+        while ((line = reader.readLine()) != null) {
+
+            // skip empty lines
+            if (line.equals("")) {
+                continue;
+            }
+            if (line.contains("Id,Name")) {
```
Comment (Raziel-777) [inline java/src/production/ConvertCsvInList.java]: You can refacto in one line these two conditions

File: java/src/production/ConvertCsvInList.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                continue;
+            }
+            if (line.startsWith(",")) {
+                line = manualId + line;
```
Comment (Raziel-777) [inline java/src/production/ConvertCsvInList.java]: You add an integer with a string ?

File: java/src/production/ConvertCsvInList.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            }
+            manualId++;
+
+            List<String> smallList = new ArrayList<>();
+
+            Scanner scanner = new Scanner(line);
+            scanner.useDelimiter(",(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)");
```
Comment (Raziel-777) [inline java/src/production/ConvertCsvInList.java]: Do you manage all whitespace between words but not in quote sentence ?

File: java/src/production/Country.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
+package production;
+
+public class Country {
```
Comment (Raziel-777) [inline java/src/production/Country.java]: You can maybe use a constructor ?

File: java/src/production/Country.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    public String getCode() {
+        return code;
+    }
+
+    public void setCode(String code) {
+        this.code = code;
+    }
+
+    @Override
+    public String toString() {
```
Comment (Raziel-777) [inline java/src/production/Country.java]: Very complicated return for this toString override method ?

File: java/src/production/ConvertCsvInList.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        while ((line = reader.readLine()) != null) {
+
+            // skip empty lines
+            if (line.equals("")) {
+                continue;
+            }
+            if (line.contains("Id,Name")) {
+                continue;
+            }
+            if (line.startsWith(",")) {
```
[the remark this answers is not in the document]
Comment (Tabardelli) [reply java/src/production/ConvertCsvInList.java]: My goal is that whatever file comes in, our output is the same.

File: java/src/production/ConvertCsvInList.java (hunk 1 above)
Comment (Tabardelli) [reply java/src/production/ConvertCsvInList.java]: Yes, we are re-creating a correct csv line so the next steps of the program can continue as usual.

File: java/src/production/Country.java (hunk 5 above)
Comment (Tabardelli) [reply java/src/production/Country.java]: Well .True. On another hand, this is the default toString generated by intellij...

File: java/src/production/ConvertCsvInList.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class ConvertCsvInList {
+
+    public ConvertCsvInList() {
+    }
+
+    public List<List<String>> readCsv(String wantedPath) throws IOException {
```
[the remark this answers is not in the document]
Comment (Tabardelli) [reply java/src/production/ConvertCsvInList.java]: Correct.

**Comment by Tabardelli** [comment]
Thanks. Melissa and I we bring little adjustments following some of your comments. 
The directory ".idea" and some ".iml" files are shown in the git history has we push them by mistake and the first stages but we corrected this before making the pull request.
And sorry for the double reader. One of them is just a first version that we should have deleted before making the pull request.

**Comment by mbaechler** [comment]
Good work

**Pull request closed without merging**